### PR TITLE
Updated wallpaper to v4.2

### PR DESCRIPTION
**Pull request by rawnly**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/wallpaper/index.d.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
-// Type definitions for wallpaper 2.5
-// Project: https://github.com/sindresorhus/wallpaper#readme
-// Definitions by: BendingBender <https://github.com/BendingBender>
+// Type definitions for wallpaper 4.2
+// Project: https://github.com/sindresorhus/wallpaper
+// Definitions by:
+//      BendingBender <https://github.com/BendingBender>
+//      Federico Vitale <https://github.com/Rawnly>
```
Comment (BendingBender) [inline types/wallpaper/index.d.ts]: This part should look like this:
```
// Definitions by: BendingBender <https://github.com/BendingBender>
//                 Federico Vitale <https://github.com/Rawnly>
```

File: types/wallpaper/index.d.ts [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export function get(): Promise<string>;
-export function set(imagePath: string, options?: WallpaperOptions): Promise<void>;
+export function get(options?: WallpaperGetOptions): Promise<string>;
+export function set(imagePath: string, options?: WallpaperSetOptions): Promise<void>;
+export function screens(): string[]
```
Comment (BendingBender) [inline types/wallpaper/index.d.ts]: this should return a `Promise<string[]>` and should be optional:
```typescript
export const screens: ScreensFn | undefined;

type ScreensFn = () => Promise<string[]>;
```

File: types/wallpaper/wallpaper-tests.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import wallpaper = require('wallpaper');
 
 wallpaper.set('unicorn.jpg').then(() => {});
-wallpaper.set('unicorn.jpg', {scale: 'fill'}).then(() => {});
+wallpaper.set('unicorn.jpg', { scale: 'fill' }).then(() => {});
 
 wallpaper.get().then(imagePath => {
```
Comment (BendingBender) [inline types/wallpaper/wallpaper-tests.ts]: Would be great if you could add some tests for `get()` with options and for `screens()`.

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: types/wallpaper/wallpaper-tests.ts (hunk 3 above)
Comment (Rawnly) [reply types/wallpaper/wallpaper-tests.ts]: Sure!

**Pull request closed without merging**